src/ripple/consensus/Consensus.h

```diff
@@ -1196,6 +1202,202 @@ void Consensus<Derived, Traits>::adjustCount (TxSet_t const& txSet,
             it.second.setVote (pit, setHas);
     }
 }
+
+/** How many of the participants must agree to reach a given threshold?
+
+    Note that the number may not precisely yield the requested percentage.
+    For example, with with size = 5 and percent = 70, we return 3, but
+    3 out of 5 works out to 60%. There are no security implications to
+    this.
+
+    @param participants the number of participants (i.e. validators)
+    @param the percent that we want to reach
+
+    @return the number of participants which must agree
+*/
+inline int
+participantsNeeded (int participants, int percent)
+{
+    int result = ((participants * percent) + (percent / 2)) / 100;
+
+    return (result == 0) ? 1 : result;
+}
+
+template <class Derived, class Traits>
+void Consensus<Derived, Traits>::updateOurPositions ()
+{
+    // Compute a cutoff time
+    auto peerCutoff = now_ - PROPOSE_FRESHNESS;
+    auto ourCutoff = now_ - PROPOSE_INTERVAL;
+
+    // Verify freshness of peer positions and compute close times
+    std::map<NetTime_t, int> closeTimes;
+    {
+        auto it = peerProposals_.begin ();
+        while (it != peerProposals_.end ())
+        {
+            if (it->second.isStale (peerCutoff))
+            {
+                // peer's proposal is stale, so remove it
+                auto const& peerID = it->second.nodeID ();
+                JLOG (j_.warn())
+                    << "Removing stale proposal from " << peerID;
+                for (auto& dt : disputes_)
+                    dt.second.unVote (peerID);
+                it = peerProposals_.erase (it);
+            }
+            else
+            {
+                // proposal is still fresh
+                ++closeTimes[effectiveCloseTime(it->second.closeTime(),
+                    closeResolution_, previousLedger_.closeTime())];
+                ++it;
+            }
+        }
+    }
+
+    // This will stay unseated unless there are any changes
+    boost::optional <TxSet_t> ourNewSet;
+
+    // Update votes on disputed transactions
+    {
+        boost::optional <TxSet_t> changedSet;
```

Comment:
You are correct.  This was due to incomplete refactoring on my part.  Previously, there was a distinct `MutableTxSet_t` type that complemented `TxSet_t` and provided a type-restricted interface for adding/removing transactions.  So `changedSet` was of type `boost::optional<MutableTxSet_t>` and was need to do the type transformation.

I ended up simplifying and adding the mutating capability to `TxSet_t` instead but forgot to simplify this spot.